include/transmem/transmem.h

```diff
@@ -153,73 +193,78 @@ friend class GraphMLWriter;
+     * @param srcFrame Identifier of the source frame.
+     * @param destFrame Identifier of the destination frame.
+     * @param tstamp Returns at what time the best transformation was available.
+     * @return A StampedAndRatedTransformation object encoding the requested transformation and additional quality information.
      * @throws InvalidArgumentException
      * @throws NoSuchLinkFoundException
      */
-    QMatrix4x4 getBestLink(const FrameID &srcFrame, const FrameID &destFrame, Timestamp &tstamp) const;
+    StampedAndRatedTransformation getBestLink(const FrameID &srcFrame, const FrameID &destFrame);
 
-    QMatrix4x4 getBestLinkCached(const FrameID &srcFrame, const FrameID &destFrame, Timestamp &tstamp);
+    /**
+     * @fn void updateLinkConfidence(const FrameID &srcFrame, const FrameID &destFrame, const double &confidence)
+     * @brief Updates the link from @a srcFrame to @a destFrame with @a confidence. If there was never such a link
+     * registered before, a NoSuchLinkFoundException is thrown.
+     * @param srcFrame Identifier of the source frame.
+     * @param destFrame Identifier of the destination frame.
+     * @param confidence Confidence value.
+     * @throws NoSuchLinkFoundException
+     */
+    void updateLinkConfidence(const FrameID &srcFrame, const FrameID &destFrame, const double &confidence);
 
+    /**
+     * @fn voidAsJSON() const
+     * @brief Dumps the current state of the transmem as JSON to a file.
+     */
     void dumpAsJSON() const;
 
+    /**
+     * @fn void dumpAsGraphML() const
+     * @brief Dumps the current graph build by the different registered links as GraphML to a file.
+     */
     void dumpAsGraphML() const;
 
 protected:
 
     bool shortestPath(Path &p) const;
-
-    bool calculateBestPointInTime(Path &path, Timestamp &tStampBestPointInTime) const;
-
-    bool calculateTransformation(const Path &path, StampedTransformation &e) const;
+    bool bestLink(StampedAndRatedTransformation &stT, Path &p) const;
+    void calculateBestPointInTime(Path &path, Timestamp &bestPoint) const;
+    void calculateTransformation(const Path &path, StampedAndRatedTransformation &resultT) const;
+    void registerLink(const FrameID &srcFrame, const FrameID &destFrame, const Timestamp &tstamp,
+                      const QQuaternion &qrot, const QQuaternion &qtrans,
+                      const double &confidence, const bool &updateConfidence);
 
     std::unordered_map<FrameID, Frame> frameID2Frame;
 
     std::deque<Link> links;
 
     std::unordered_map< std::string, std::pair< Timestamp, Path > > cachedBestLinks;
-    std::unordered_map< std::string, QMatrix4x4 > cachedBestTransformations;
+    std::unordered_map< std::string, StampedAndRatedTransformation > cachedBestTransformations;
 
     DurationSec storageTime{10};
 
+    const double defaultLinkConfidence = 1.;
```

Comment:
I don't understand the question. The default link confidence can already be set via an extra constructor?